validate/labels.go

```diff
@@ -46,7 +51,11 @@ func (l *AllowedLabels) Validate(lbls map[string]string) error {
 		violations = append(violations, l.ValidateLabel(k, v))
 	}
 
-	return multierr.Combine(violations...)
+	if err := multierr.Combine(violations...); err != nil {
+		return fmt.Errorf("label validation failed: %w, allowed labels: %s", err, l)
```

Comment:
This will not result in a particularly nice message, since it will have something like `allowed labels: allowed { ... },`. I'd probably add a separate method on `AllowedLabels` which only returns a formatted list of labels without the internal field identifier.